lib/pf/pki_provider/packetfence_pki.pm

```diff
@@ -87,9 +100,25 @@ Revoke the certificate for a user
 sub revoke {
     my ($self, $cn) = @_;
     my $logger = get_logger();
+    my %options = (
+        -where => {
+            id => "/pki/$cn",
+        }
+    );
+
+    my ($status, $iter) = pf::dal::key_value_storage->search(%options);
+    if (is_success($status)) {
+        my $key_value = $iter->next(undef);
+        if ($key_value) {
```

Comment:
Would it be relevant to log an error or warning when we don't find the value?

It shouldn't happen if a cert issued by this code gets revoked but will happen for certs issued by the previous code from my understanding